physics/GFS_rrtmgp_cloud_mp.F90

```diff
@@ -0,0 +1,883 @@
+         cld_resnow,                & ! Effective radius for              snow   hydrometeors
+         cld_rwp,                   & ! Water path for                    rain   hydrometeors
+         cld_rerain,                & ! Effective radius for              rain   hydrometeors
+         precip_frac,               & ! Precipitation fraction
+         cld_cnv_frac,              & ! Cloud-fraction for   convective clouds
+         cld_cnv_lwp,               & ! Water path for       convective   liquid cloud-particles
+         cld_cnv_reliq,             & ! Effective radius for convective   liquid cloud-particles
+         cld_cnv_iwp,               & ! Water path for       convective   ice    cloud-particles
+         cld_cnv_reice,             & ! Effective radius for convective   ice    cloud-particles
+         cld_pbl_lwp,               & ! Water path for       SGS PBL liquid cloud-particles
+         cld_pbl_reliq,             & ! Effective radius for SGS PBL liquid cloud-particles
+         cld_pbl_iwp,               & ! Water path for       SGS PBL ice    cloud-particles
+         cld_pbl_reice                ! Effective radius for SGS PBL ice    cloud-particles
+    character(len=*), intent(out) :: &
+         errmsg                       ! Error message
+    integer, intent(out) :: &  
+         errflg                       ! Error flag
+
+    ! Local
+    integer :: iCol, iLay
+    real(kind_phys) :: alpha0
+    real(kind_phys), dimension(nCol,nLev) :: cldcov, cldtot, cldcnv
+
+    if (.not. (doSWrad .or. doLWrad)) return
+
+    ! Initialize CCPP error handling variables
+    errmsg = ''
+    errflg = 0
+
+    ! ###################################################################################
+    ! GFDL Microphysics
+    ! ("Implicit" SGS cloud-coupling to the radiation)
+    ! ###################################################################################
+    if (imp_physics == imp_physics_gfdl) then
+       ! GFDL-Lin
+       if (.not. lgfdlmprad) then
+          errflg = 1
+          errmsg = "ERROR: MP choice not available with RRTMGP"
+          return
+       ! GFDL-EMC
+       else
+
+          ! "cld_frac" is modified prior to include subgrid scale cloudiness, see 
+          ! module_SGSCloud_RadPre.F90.
+          do iLay = 1, nLev
+             do iCol = 1, nCol
+                ! 
+                ! SGS clouds present, use cloud-fraction modified to include sgs clouds.
+                !
+                if ((imfdeepcnv==imfdeepcnv_gf .or. do_mynnedmf) .and. kdt>1) then
+                   ! MYNN sub-grid cloud fraction.
+                   if (do_mynnedmf) then
+                      ! If rain/snow present, use GFDL MP cloud-fraction...
+                      if (tracer(iCol,iLay,i_cldrain)>1.0e-7 .OR. tracer(iCol,iLay,i_cldsnow)>1.0e-7) then
+                         cld_frac(iCol,iLay) = tracer(iCol,iLay,i_cldtot)
+                      endif
```

Comment:
This MYNN portion (lines 183 to about 190 and the associated endif) should be removed.

Reply:
According to Sam's comments below, we aren't supposed to make changes, but as soon as we are allowed to make changes, this part of the code will need to be removed.

Reply:
Is this a critical bug that needs to be fixed in the community repository? Or can we just change it in gsl/develop and worry about the community in a couple months?

Reply:
It will only cause problems when we start testing our suite with RRTMGP, so no, it's not immediately urgent. It may take considerable time to go through all of the radiation-related changes and find a clean fix.